src/ChannelDetector.ts

```diff
@@ -645,35 +657,40 @@ export class ChannelDetector {
             });
 
             _keysOptional = [
-                ...Array.from(deviceObjs.values()).sort(),
```

Comment:
I think sorting the arrays one by one was intentional by @Apollon77. If all ids are in one huge array and sorted, device, channel, folder and state ids are mixed. But Ingo wanted to have devices before channels before folders before other objects.
Not 100% sure, but that's what he did in his original commit https://github.com/ioBroker/ioBroker.type-detector/commit/b6b71d51890c833ddf18156dc2c1912244c6ffa2

Reply:
I analysed the code and could not find any place where it is required in such a form

Reply:
It is an optimization though for mass detection to make sure to check all devices and channels first. Rest is then only checked if not already mapped anywhere else

Reply:
But in fact it is "just" an optimization I thought about ... if we control in any case from outside how we search and which ID is checked then this might also be better "outside" of the detect function when doing mass searches?

Reply:
But if we do not do that then also the whole "sort objects into types and re-join them " can also be killed and we can just do a dump global sort as it was before

Reply:
It is killed, as not used